src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,64 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { closeСurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, [currentTodo]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
-
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(closeСurrentTodo())}
+            />
+          </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
+            <p className="block" data-cy="modal-user">
+              {currentTodo?.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${user?.email}}`}>{user?.name}</a>
```

Comment:
This line assumes that `user` is defined and it has `email` and `name` properties. It's better to add some checks before using these properties to prevent possible errors if `user` is `undefined` or doesn't have these properties.